src/ripple/app/misc/tests/CanonicalTXSet.test.cpp

```diff
@@ -0,0 +1,158 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012, 2013 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/app/misc/CanonicalTXSet.h>
+#include <ripple/basics/TestSuite.h>
+
+namespace ripple {
+namespace test {
+
+class CanonicalTXSet_test : public TestSuite
+{
+    void test_txset_key_ordering()
+    {
+        // Verify that CanonicalTXSet::Key sorts as expected.
+        using Key = CanonicalTXSet::Key;
```

Comment:
You could eliminate the key entirely:
```
// A StrictWeakOrdering for putting
// transactions in a canonical ordering
class CanonicalLess
{
private:
    ...
public:
    CanonicalLess (uint256 const& lastClosedLedgerDigest);
    template <class Object>
    bool operator()(std::shared_ptr<Object const> const& lhs,
        std::shared_ptr<Object const> const& rhs) const;
};
```

Now you can replace the member in CanonicalTxSet with `std::set<std::shared_ptr<STTx const>, CanonicalLess>` and still have access to the sort for your unit tests (for example, by using `std::sort` on a vector of transactions.